cub/cub/agent/agent_three_way_partition.cuh

```diff
@@ -47,6 +47,76 @@ CUB_NAMESPACE_BEGIN
  * Tuning policy types
  ******************************************************************************/
 
+namespace detail
+{
+
+namespace three_way_partition
+{
+
+template <class OffsetT>
+struct pair_pack_t
+{
+  OffsetT x, y;
+
+  __device__ pair_pack_t<OffsetT> operator+(const pair_pack_t<OffsetT> &other) const
+  {
+    return {x + other.x, y + other.y};
+  }
+};
+
+template <class OffsetT, class = void>
+struct accumulator_pack_base_t
+{
+  using pack_t = pair_pack_t<OffsetT>;
+
+  __device__ static pack_t pack(OffsetT f, OffsetT s) { return {f, s}; }
+  __device__ static OffsetT first(pack_t packed) { return packed.x; }
+  __device__ static OffsetT second(pack_t packed) { return packed.y; }
+};
+
+template <class OffsetT>
+struct accumulator_pack_base_t<OffsetT, typename cuda::std::enable_if<sizeof(OffsetT) == 4>::type>
```

Comment:
I assume you have experimented with 64 bit accumulators and they did not help with the slowdown?

Reply:
When `OffsetT` is 32 bit, the accumulator is 64 bit. If you mean experimenting with 64 bit offsets, there's little that can be done without https://github.com/NVIDIA/cccl/issues/220.